digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/NodePointDAO.scala

```diff
@@ -338,9 +341,104 @@ class NodePointDAO extends BaseDAO {
     else
       Q.updateNA(query).first
   }
+  
+  def expireByNodeNumberAndType(node_number: Long, typeId: Long): Unit = {
+    val query =
+      s"""
+        Update NODE_POINT Set valid_to = sysdate where valid_to IS NULL AND node_number = $node_number
+        AND type = $typeId
+      """
+    Q.updateNA(query).execute
+  }
```

Comment:
Didn't we were suppose to remove all previously calculated node points? Instead of saving them as history?

Reply:
valid_from and valid_to are modification history -fields, not history-fields like start_date and end_date. If valid_to is set, the row is logically deleted. We don't really delete rows, we just expire them by setting valid_to = sysdate.